src/config/system.rs

```diff
@@ -4,30 +4,241 @@ pub const WS_SUB_MANAGER_ID: u32 = 2;
+    }
+}
 
 #[cfg(feature = "journald")]
 pub fn log_journald(level: u32, message: &str) {
     use systemd::journal;
     journal::print(level, message);
 }
 
+// #[cfg(feature = "prometheusd")]
+#[derive(Debug)]
+pub enum MetricsCommand<'a> {
+    Flush(),
+    Channel(&'a MetricSender, &'a MetricReceiver),
+    PushLatency(&'a RpcMetrics, &'a RegistryChannel, &'a str, &'a str, f64),
+    PushRequest(
+        &'a RpcMetrics,
+        &'a RegistryChannel,
+        &'a str,
+        &'a str,
+        &'a u16,
+        Duration,
+    ),
+    PushError(
+        &'a RpcMetrics,
+        &'a RegistryChannel,
+        &'a str,
+        &'a str,
+        &'a u16,
+        Duration,
+    ),
+    Push(&'a RpcMetrics, &'a RegistryChannel),
+    Pull(&'a RpcMetrics, &'a RegistryChannel),
+}
+
+// #[cfg(feature = "prometheusd")]
+#[derive(Debug)]
+pub enum MetricChannelCommand {
+    AdminMsg(oneshot::Sender<RpcMetrics>),
+    StatsMsg(oneshot::Sender<RpcMetrics>),
+}
+
+//#[cfg(feature = "prometheusd")]
+pub async fn metrics_update_sink(mut metrics_rx: MetricReceiver) {
+    loop {
+        while metrics_rx.recv().await.is_some() {
+            continue;
+        }
+    }
+}
+//#[cfg(feature = "prometheusd")]
+pub async fn listen_for_metrics(_metric: RpcMetrics, _metrics_tx: MetricSender) {
+    unimplemented!()
+    // let mut tx.inner.
+}
+
+pub async fn metrics_listener(
+    mut metrics_rx: MetricReceiver,
+    metrics_status: Arc<RwLock<RpcMetrics>>,
+) {
+    while let Some(update) = metrics_rx.recv().await {
+        //TODO: match for metrics type, latency, errors, requests
+        //match metric {
+        //MetricsType::Latency => {
+        //}
+        let mut metrics = metrics_status.write().unwrap();
+        *metrics = update;
+    }
+}
+
+async fn metrics_processor(
+    mut metrics_rx: MetricReceiver,
+    registry_state: Arc<RwLock<StorageRegistry>>,
+    // path: &str,
+    // method: &str,
+    // status: &u16,
+    // duration: Duration,
+) {
+    use crate::log_info;
+    let _metrics = RpcMetrics::init(&registry_state.read().unwrap()).unwrap();
+    let mut interval = tokio::time::interval(Duration::from_millis(10));
+    loop {
+        interval.tick().await;
+        while let Some(incoming) = metrics_rx.recv().await {
+            incoming.requests_complete("test", "test", &200, Duration::from_millis(100));
+            let test_report = metrics_encoder(registry_state.clone()).await;
+            log_info!("prometheus metrics: {:?}", test_report);
+            let _registry = registry_state.read().unwrap();
+            // incoming.requests_complete(path, method, status, duration);
+        }
+    }
+}
+/// Accepts metrics request, encodes and prints
+//#[cfg(feature = "prometheusd")]
+async fn accept_metrics_request(
+    tx: Request<hyper::body::Incoming>,
+    metrics_tx: Arc<MetricSender>,
+    registry_state: Arc<RwLock<StorageRegistry>>,
+) -> Result<String, Infallible> {
+    if tx.uri().path() == "/ws_metrics" {
```

Comment:
idk if we have the metrics separated or together? 

for regular requests idk considering you're serving similar data, for subscriptions that are WS only maybe???